exporter/datadogexporter/internal/hostmetadata/provider/provider.go

```diff
@@ -21,16 +21,44 @@ type chainProvider struct {
 	priorityList []string
 }
 
+type providerReply struct {
```

Comment:
Since this type isn't used anywhere else outside `Source`, you can move the type definition inside the function (and call it `reply`).

Reply:
Good point, moved